### ISSUE TITLE
ngx-pcre-stats crashes

### ISSUE THREAD

**Issue by dailei** [openresty/nginx-systemtap-toolkit#23]
大神，我英文不太好，就直接使用中文了，请见谅
我在使用这个工具的时候，报下面的错误
我想知道这是因为缺少相关的依赖吗，还是这个工具不支持LuaJIT2.1，或者是其它的原因？
谢谢

    [root@srv-02 src]# uname -a
    Linux srv-02.sc 3.10.0-123.13.2.el7.x86_64 #1 SMP Thu Dec 18 14:09:13 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
    [root@srv-02 src]# ~/nginx-systemtap-toolkit/ngx-pcre-stats -p 12897 --worst-time-top --luajit20
    semantic error: unable to find member 'fr' for struct lua_TValue (alternatives: tt, value): operator '->' at <input>:32:93
            source:     gcobj = gcref(&@cast(o, "TValue", "/usr/local/openresty-1.7.7.1-debug/nginx/sbin/nginx")->fr->func)
                                                                                                                ^

    semantic error: type definition 'GCobj' not found in '/usr/local/openresty-1.7.7.1-debug/nginx/sbin/nginx': operator '@cast' at :33:13
            source:     return &@cast(gcobj, "GCobj", "/usr/local/openresty-1.7.7.1-debug/nginx/sbin/nginx")->fn
                                ^

    Pass 2: analysis failed.  [man error::pass2]
    Number of similar error messages suppressed: 1.
    Rerun with -v to see them.

**Comment by agentzh** [comment]
@dailei `ngx-pcre-stats` does support LuaJIT 2.1 when the `--luajit20` option is specified. I've just tested your usage locally and it works fine.

From the error messages you provided, it seems that your OpenResty is actually built with the standard Lua 5.1 interpreter instead of (any versions of) LuaJIT. You can try replacing the `--luajit20` option with the `--lua51` option in your `ngx-pcre-stats` command line.

You can further check whether you're using LuaJIT or the standard Lua 5.1 interpreter by the methods listed below:

http://stackoverflow.com/questions/19056429/how-to-check-if-nginx-uses-luajit-and-not-lua

Basically OpenResty uses its bundled version of LuaJIT by default unless you explicitly specify the `--with-lua51` or `--with-lua51=DIR` options on the `./configure` command line while building OpenResty.

BTW, if you really want to use Chinese, please join the openresty (Chinese) mailing list instead. See http://openresty.org/#Community Thanks for your cooperation!

**Comment by dailei** [comment]
@agentzh I checked my `./configure` options, I carelessly specified `--with-lua51` option while building OpenResty.

I  rebuilt my  OpenResty without `--with-lua51` option, the problem has been resolved.

Thanks for your help. Once again apologize for my carelessness.
